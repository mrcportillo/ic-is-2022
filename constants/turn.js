export const TURN_TYPES = {
  VIP: "VIP",
  GENERAL: "GENERAL",
};

export const TURN_WAITING_TIME_BOUNDARIES = {
  GOOD: 10,
  MEDIUM: 20,
};
