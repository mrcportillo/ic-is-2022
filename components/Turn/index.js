import { getTurnBorderColorByWaitingTime } from "../../libs/turn";
import styles from "../../styles/styles.module.css";

function Turn({ turn, loading }) {
  return (
    <div
      className={
        loading ? `${styles.turn_card} ${styles.loading}` : styles.turn_card
      }
      style={{
        borderColor: getTurnBorderColorByWaitingTime(turn?.estimated_minutes),
      }}
    >
      <div className={styles.turn_header}>
        <p>{turn?.type} TURN</p>
        <p># {turn?.position}</p>
      </div>
      <p>Date: {turn?.date && new Date(turn.date).toDateString()}</p>
      <p>People before: {turn?.positions_before}</p>
      <p>Waiting time: {turn?.estimated_minutes} minutes</p>
    </div>
  );
}

export default Turn;
