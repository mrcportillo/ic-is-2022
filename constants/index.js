import { TURN_TYPES } from "./turn";

export { TURN_TYPES };
