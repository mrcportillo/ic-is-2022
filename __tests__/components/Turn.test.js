import renderer from "react-test-renderer";
import Turn from "../../components/Turn";
import { TURN_TYPES } from "../../constants";

describe("Turn component", () => {
  const MOCKED_TURN = {
    type: TURN_TYPES.GENERAL,
    position: 1,
    date: "Tue, 31 May 2022 17:21:34 GMT",
    positions_before: 0,
    estimated_minutes: 2,
  };
  test("Loading turn", () => {
    const component = renderer.create(<Turn loading />).toJSON();
    expect(component).toMatchSnapshot();
  });
  test("Showing turn", () => {
    const component = renderer.create(<Turn turn={MOCKED_TURN} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
