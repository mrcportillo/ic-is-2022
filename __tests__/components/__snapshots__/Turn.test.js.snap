// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Turn component Loading turn 1`] = `
<div
  className="turn_card loading"
  style={
    Object {
      "borderColor": "firebrick",
    }
  }
>
  <div
    className="turn_header"
  >
    <p>
       TURN
    </p>
    <p>
      # 
    </p>
  </div>
  <p>
    Date: 
  </p>
  <p>
    People before: 
  </p>
  <p>
    Waiting time: 
     minutes
  </p>
</div>
`;

exports[`Turn component Showing turn 1`] = `
<div
  className="turn_card"
  style={
    Object {
      "borderColor": "green",
    }
  }
>
  <div
    className="turn_header"
  >
    <p>
      GENERAL
       TURN
    </p>
    <p>
      # 
      1
    </p>
  </div>
  <p>
    Date: 
    Tue May 31 2022
  </p>
  <p>
    People before: 
    0
  </p>
  <p>
    Waiting time: 
    2
     minutes
  </p>
</div>
`;
